go/kbfs/libkbfs/favorites.go

```diff
@@ -342,7 +350,40 @@ func (f *Favorites) handleReq(req *favReq) (err error) {
 	// quickly when offline.
 	needFetch := (req.refresh || f.favCache == nil) && !req.clear
 	wantFetch := f.config.Clock().Now().After(f.cacheExpireTime) && !req.clear
-	changed := false
+
+	for _, fav := range req.toAdd {
+		// FOR DISCUSSION: we don't check this in toDel because it could be
+		// out-of-date. So perhaps we should get rid of this too? Relevant
+		// question, if a favorite is deleted from another device, do we get
+		// notification from core for cache invalidation?
+		_, present := f.favCache[fav.Folder]
+		if !fav.Created && present {
+			continue
+		}
+		//
```

Comment:
Question for @keybase/kbfs-hackers in the comment above.

Reply:
I believe this check for adds is critical and we should definitely leave it in.  We've had issues in the past with spamming the API server with adding the same favorite multiple times.  We don't have the same problem with deletes, because after the user deletes it, they aren't accessing the folder again.  But with adds, we could be going through this code on basically every folder access.

Yes, favorite deletes from another device result in a notification to this device.

Reply:
Sounds good. I was mostly worried about the races. But with notifications it seems the race is not possible. Will steal your response here as code comment.